python/cudf/cudf/core/series.py

```diff
@@ -2140,39 +2140,79 @@ def label_encoding(self, cats, dtype=None, na_sentinel=-1):
         Parameters
         ----------
         values : sequence of input values
-        dtype: numpy.dtype; optional
-               Specifies the output dtype.  If `None` is given, the
-               smallest possible integer dtype (starting with np.int8)
-               is used.
-        na_sentinel : number
+        dtype : numpy.dtype; optional
+            Specifies the output dtype.  If `None` is given, the
+            smallest possible integer dtype (starting with np.int8)
+            is used.
+        na_sentinel : number, default -1
             Value to indicate missing category.
```

Comment:
How is this function used that we need sentinel values? I'm wondering if we should just generate a column with nulls instead.

Reply:
This seems to be introduced in this PR: https://github.com/rapidsai/cudf/pull/91/files

To me it looks like `label_encoding`s `na_sentinel` is to support `factorize`s `na_sentinel`.

But it also looks like it isn't being passed through from `factorize` to `label_encoding` 😄

Reply:
Does anywhere other than `factorize` use `label_encoding`? I'm wondering if it would make sense for us to change the default for `na_sentinel` to `None` and if it's `None` we use nulls instead of an actual sentinel value.

Reply:
Seems like we are currently asserting `base_mask` to be `None`, this assertion will throw if we introduce `null` values

Reply:
We're asserting the child's mask is `None`. Given this function only does the encoding and returns an integer dtype Series, we could return the Series with a nullmask instead of null sentinel, and then pull the nullmask up to the parent column in the case where we're constructing a CategoricalColumn.

Either way, lets not tackle this in this PR, but can you file an issue for us to discuss?

Reply:
> Does anywhere other than `factorize` use `label_encoding`? I'm wondering if it would make sense for us to change the default for `na_sentinel` to `None` and if it's `None` we use nulls instead of an actual sentinel value.

I think it is also used in converting a any other column to categorical. https://github.com/rapidsai/cudf/blob/25a6e312cdd3498b87320cdf32690eeccdd1850a/python/cudf/cudf/core/column/column.py#L892

Reply:
Added a separate discussion: https://github.com/rapidsai/cudf/issues/5622